utils/mem-cache.js

```diff
@@ -0,0 +1,93 @@
+// @ts-check
+/**
+ * An in-memory cache.
+ * @template ValueType
+ */
+class MemCache {
+  #ttl = Infinity;
+  /** @type {Map<string, { time: number, value: ValueType }>} */
+  #map = new Map();
+
+  /**
+   * @param {number} ttl cache TTL in milliseconds
+   */
+  constructor(ttl) {
+    this.#ttl = ttl;
+  }
+
+  /**
+   * @param {string} key
+   * @param {ValueType} value
+   */
+  set(key, value) {
+    this._set(key, Date.now(), value);
```

Comment:
the private `._set` and `._get` seem to just add unnecessary indirection (and yucky python looking... omg so gross!). Maybe consider inlining them? 🙏

Reply:
Used them here we're using them in fs-cache also..